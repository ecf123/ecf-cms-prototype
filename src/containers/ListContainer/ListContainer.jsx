import "./ListContainer.scss";
import PageTitle from "../../components/PageTitle/PageTitle";
import SearchBar from "../../components/SearchBar/SearchBar";
import Button from "../../components/Button/Button";
import CardContainer from "../CardContainer/CardContainer";
import { ReactComponent as Pencil } from "../../assets/pen-icon.svg";
import { Link } from "react-router-dom";
import { useState } from "react";

const ListContainer = ({ title, data }) => {

  const cards=data?.map(element=>{
    if(title == "Article"){
      const paragraphContent = element.articleContent[0].paragraph
      return{
        title:element.title,
        img: element.thumbnail,
        cardInfo:paragraphContent.substr(0, Math.min(paragraphContent.length, paragraphContent.indexOf(".",50)+1)),
        dateOrTime: element.date,
        links: element.category.split(",")
      }
    } else {
      return {
        title:element.heading,
        img: element.image,
        cardInfo: element.overview.substr(0, Math.min(element.overview.length, element.overview.indexOf(".",50)+1)),
        links: [element.skillPoints,element.trophies],
      }
    }
  })

  const [searchTerm, setSearchTerm] = useState("");

  const condition = title == "Article" ? false : true;

  const handleSearchInput = (event) => {
    setSearchTerm(event.target.value);
  };

  const filteredCards = cards?.filter((card) =>
    card.title.toLowerCase().includes(searchTerm.toLowerCase())
  );

  return (
    <div className="list-page">
      <div className="list-page__header">
        <PageTitle
          className="list-page__title page-title"
          title={title + " List"}
        />
      </div>
      <div className="list-page__body">
        <nav className="list-page__nav">
          <SearchBar
            placeHolderText={"Search for " + title}
            handleSearchInput={handleSearchInput}
            value={searchTerm}
          />
          <Link to="*">
            <Button
              style={"button black round-border large"}
              imgStyle={"medium-img white-svg"}
              textStyle={"text medium-text black"}
              buttonText={"Add " + title}
            >
              <Pencil />
            </Button>
          </Link>
        </nav>
        <CardContainer cards={filteredCards} condition={condition}/>
      </div>
    </div>
  );
};

export default ListContainer;
