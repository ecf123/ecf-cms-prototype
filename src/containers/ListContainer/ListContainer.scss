@use "../../styles/variables.scss" as *;


.list-page{

  &__header{
    border-bottom: 2px solid $color-grey-light;
    padding: 3rem 0;;
  }

  &__title{
    padding-left: 3rem;
  }

  &__body{
    padding: 0 3rem;
  }

  &__nav{
    display: flex;
    justify-content: space-between;
    margin-top: 3rem;
  }
}