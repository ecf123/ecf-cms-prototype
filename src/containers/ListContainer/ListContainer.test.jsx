import ListContainer from "./ListContainer";
import { render, screen } from "@testing-library/react";
import { customRender } from "../../utils/testUtils";
import image from "../../assets/for-hire-image.svg";
import userEvent from "@testing-library/user-event" 


const cards = [
    {
      img: image,
      dateOrTime: "12:01",
      title: "Test title one",
      cardInfo: "Test text one",
      links: ["one", "two", "three"],
    },
    {
      img: image,
      dateOrTime: "12:00",
      title: "Test title two",
      cardInfo: "Test text two",
      links: ["one", "two", "three"],
    },
  ];



it("it should match the snapshot without props", () => {
    const { container } = customRender(<ListContainer />);
    expect(container).toMatchSnapshot();
  });


  it("it should match the snapshot with props", () => {
    const { container } = customRender(<ListContainer cards={cards} title="title"/>);
    expect(container).toMatchSnapshot();
  });


  it("should render the title on the screen", () => {
    customRender(<ListContainer title="Title"/>);
    const title = screen.getByText("Title List");

    expect(title).toBeInTheDocument();
  });

  it("should render the input box on the screen", () => {
    customRender(<ListContainer />);
    const input = screen.getByRole("textbox");

    expect(input).toBeInTheDocument();
  })

  it("should render the 'add course' button on the screen", () => {
    customRender(<ListContainer />);
    const button = screen.getByRole("button");

    expect(button).toBeInTheDocument();
  })

  it("should display the first card if search 'test title one'", () => {
    customRender(<ListContainer cards={cards}/>)

    const searchCard1 = screen.getByText("Test title one");
    const searchCard2 = screen.getByText("Test title two");

    expect(searchCard1).toBeInTheDocument();
    expect(searchCard2).toBeInTheDocument();

    const searchBox = screen.getByRole("textbox");
    userEvent.type(searchBox, "one");
    const searchBox1 = screen.getByText("Test title one");
    const searchBox2 = screen.queryByText("Test title two");

    expect(searchBox1).toBeInTheDocument();
    expect(searchBox2).toBeFalsy();
  })





