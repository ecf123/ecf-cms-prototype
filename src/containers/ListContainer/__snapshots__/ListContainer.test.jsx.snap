// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should match the snapshot with props 1`] = `
<div>
  <div
    class="list-page"
  >
    <div
      class="list-page__header"
    >
      <h1
        class="list-page__title page-title"
      >
        title List
      </h1>
    </div>
    <div
      class="list-page__body"
    >
      <nav
        class="list-page__nav"
      >
        <div
          class="search-bar"
        >
          <img
            alt="Search Icon"
            class="search-bar__image"
            src="magnifying-glass-grey.svg"
          />
          <input
            class="search-bar__input"
            placeholder="Search for title"
            type="text"
            value=""
          />
        </div>
        <a
          href="/*"
        >
          <button
            class="button black round-border large"
          >
            <div
              class="medium-img white-svg"
            >
              <svg>
                pen-icon.svg
              </svg>
            </div>
            <p
              class="text medium-text black"
            >
              Add title
            </p>
          </button>
        </a>
      </nav>
      <div
        class="card-container"
      >
        <div
          class="card"
        >
          <img
            alt="Card Image"
            class="card__image"
            src="for-hire-image.svg"
          />
          <p
            class="card__date"
          >
            12:01
          </p>
          <div
            class="card__header"
          >
            <h2
              class="card__title"
            >
              Test title one
            </h2>
            <img
              alt="Arrow Icon"
              class="card__icon"
              src="arrow-upper-right-black.svg"
            />
          </div>
          <p
            class="card__info"
          >
            Test text one
          </p>
          <ul
            class="card__links"
          >
            <li
              class="card__link card__link--purple"
            >
              one
            </li>
            <li
              class="card__link card__link--darkpurple"
            >
              two
            </li>
            <li
              class="card__link card__link--purple"
            >
              three
            </li>
          </ul>
        </div>
        <div
          class="card"
        >
          <img
            alt="Card Image"
            class="card__image"
            src="for-hire-image.svg"
          />
          <p
            class="card__date"
          >
            12:00
          </p>
          <div
            class="card__header"
          >
            <h2
              class="card__title"
            >
              Test title two
            </h2>
            <img
              alt="Arrow Icon"
              class="card__icon"
              src="arrow-upper-right-black.svg"
            />
          </div>
          <p
            class="card__info"
          >
            Test text two
          </p>
          <ul
            class="card__links"
          >
            <li
              class="card__link card__link--purple"
            >
              one
            </li>
            <li
              class="card__link card__link--darkpurple"
            >
              two
            </li>
            <li
              class="card__link card__link--purple"
            >
              three
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`it should match the snapshot without props 1`] = `
<div>
  <div
    class="list-page"
  >
    <div
      class="list-page__header"
    >
      <h1
        class="list-page__title page-title"
      >
        undefined List
      </h1>
    </div>
    <div
      class="list-page__body"
    >
      <nav
        class="list-page__nav"
      >
        <div
          class="search-bar"
        >
          <img
            alt="Search Icon"
            class="search-bar__image"
            src="magnifying-glass-grey.svg"
          />
          <input
            class="search-bar__input"
            placeholder="Search for undefined"
            type="text"
            value=""
          />
        </div>
        <a
          href="/*"
        >
          <button
            class="button black round-border large"
          >
            <div
              class="medium-img white-svg"
            >
              <svg>
                pen-icon.svg
              </svg>
            </div>
            <p
              class="text medium-text black"
            >
              Add undefined
            </p>
          </button>
        </a>
      </nav>
      <div
        class="card-container"
      />
    </div>
  </div>
</div>
`;
